src/coreclr/System.Private.CoreLib/src/System/Array.CoreCLR.cs

```diff
@@ -223,6 +223,22 @@ public static unsafe void Clear(Array array, int index, int length)
             // GC.KeepAlive(array) not required. pMT kept alive via `ptr`
         }
 
+        private unsafe nint GetFlattenedIndex(int rawIndex)
+        {
+            // Checked by the caller
+            Debug.Assert(Rank == 1);
+
+            if (RuntimeHelpers.GetMethodTable(this)->IsMultiDimensionalArray)
```

Comment:
Doesn't this method need KeepAlive for this at the end of this scope?

Reply:
This is always called before `InternalGetValue`, which also has a `GC.KeepAlive(this)` call at the end.
That should be enough to cover this one here too, right? 🤔